components/select/nz-select-top-control.component.ts

```diff
@@ -26,7 +26,12 @@ import { NzSelectService } from './nz-select.service';
   animations: [zoomMotion],
   changeDetection: ChangeDetectionStrategy.OnPush,
   encapsulation: ViewEncapsulation.None,
-  templateUrl: './nz-select-top-control.component.html'
+  templateUrl: './nz-select-top-control.component.html',
+  host: {
+    '[class.ant-select-selection--single]': '!nzCustomTemplate && nzSelectService.isSingleMode',
```

Comment:
does nzCustomTemplate has any relationship with single mode or multiple mode?

Reply:
no, but `ant-select-selection--single` has fixed height 32px which would break the ui with custom template, and it works fine with `ant-select-selection--multiple` class

Reply:
`ant-select-selection--multiple` and `ant-select-selection--single` styles are sync from antd react, it may change at anytime, there is no guarantee that `ant-select-selection--multiple` will always work.

Reply:
Yea I don't like this workaround neither, do you have any suggestions? maybe a new class or styles for custom template only?

Reply:
just keep the ant-select-selection--single, I think the height 32px should be kept.

Reply:
what if we have multi-line template like the example does? it breaks the ui.

Reply:
yes, it should be another pr to antd react repo, we should not change the actual meaning of className.

Reply:
sure, let me see what I can do with it.